open-sdg/sdg-build
# Allow per-indicator callbacks in Open SDG output

This feature allows Open SDG implementations to perform some custom Python code on each indicator - not for the purposes of altering any output - but for any other purpose, like maybe to generate some additional output that is not in sdg-build.

Testing instructions:

In your data repository's [build_data.py](https://github.com/open-sdg/open-sdg-data-starter/blob/develop/scripts/build_data.py) file try the following code:

```
from sdg.open_sdg import open_sdg_build

def my_indicator_callback(indicator):
  print('Running indicator callback for indicator ' + indicator.inid)

open_sdg_build(config='config_data.yml', indicator_callback=my_indicator_callback)
```

When the data build runs, there should be log entries printed that say "Running indicator callback for indicator 1.1.1" etc.

Q | A
--- | ---
Feature branch/test site URL | [Link](https://github.com/ONSdigital/sdg-data/tree/feature-per-indicator-callbacks
)
Fixed issues | N/A
Related version | 2.2.0-dev
Bugfix, feature or docs? |
